devito/compiler.py

```diff
@@ -194,12 +194,14 @@ def get_tmp_dir():
 
     :return: Path to a devito-specific tmp directory
     """
-    tmpdir = path.join(gettempdir(), "devito-%s" % getuid())
-
-    if not path.exists(tmpdir):
-        mkdir(tmpdir)
-
-    return tmpdir
+    global _devito_compiler_tmpdir
+    try:
+        path.exists(_devito_compiler_tmpdir)
+    except:
+        suffix = "devito-%s" % getuid()
+        _devito_compiler_tmpdir = mkdtemp(prefix=gettempdir(), suffix=suffix)
```

Comment:
lots of tests failing, probably because you're missing a `/` somewhere here. How could this work on your machine?